Earthfile

```diff
@@ -140,16 +140,35 @@ dist:
 docker:
     ARG FLAVOR
     ARG VARIANT
-
     FROM $BASE_IMAGE
 
+    DO +PROVIDER_INSTALL
+
+    ARG KAIROS_VERSION
+    IF [ "$KAIROS_VERSION" = "" ]
+        ARG OS_VERSION=${VERSION}
+    ELSE
+        ARG OS_VERSION=${KAIROS_VERSION}
+    END
+
+    ARG OS_ID
+    ARG OS_NAME=${OS_ID}-${FLAVOR}
+    ARG OS_REPO=quay.io/kairos/${VARIANT}-${FLAVOR}
+    ARG OS_LABEL=latest
+
+    DO kairos+OSRELEASE --BUG_REPORT_URL="https://github.com/kairos-io/kairos/issues/new/choose" --HOME_URL="https://github.com/kairos-io/provider-kairos" --OS_ID=${OS_ID} --OS_LABEL=${OS_LABEL} --OS_NAME=${OS_NAME} --OS_REPO=${OS_REPO} --OS_VERSION=${OS_VERSION}-k3s${K3S_VERSION} --GITHUB_REPO="kairos-io/provider-kairos" --VARIANT=${VARIANT}
```

Comment:
I guess earthly can handle this back and forth between repos, cool ... (starting form kairos repo) `earthly +some-target` -> `provider-kairos+PROVIDER_INSTALL` -> `kairos+OSRELEASE`

Reply:
well, kairos target will only use the PROVIDER_INSTALL command, so it doesnt trigger this, so its only some target -> kairos+OSRELEASE. Im not even sure that would work.

Maybe we should move some of the generic targets to a different repo like earthly-utils or something and have our shared commands there so we avoid recursive deps...

Reply:
yeah good idea